### PR TITLE
Move away form blacklist terminology

### PR DESCRIPTION
**Pull request by mocsy**
changelog: Move away form blacklist terminology

### CODE REVIEW

**Comment by flip1995** [comment]
**Moderating note:** I think we all know what discussions this type of PR can introduce, so I want to ask everyone to stay civil and have a productive discussion about this. (I really don't want to lock this PR)

---

cc @rust-lang/clippy 

Do we want to do this? Recently all internal uses of black/white-list were renamed in the compiler. No user facing things were renamed though. This would definitely be user facing, both the lint name and the configuration name.

We have precedent for renaming a lint, because it could be offending to some people: https://github.com/rust-lang/rust-clippy/issues/3521

**Comment by oli-obk** [comment]
Yea this definitely needs a renaming scheme, other than that lgtm

**Comment by mikerite** [comment]
It's got to be done without breaking peoples config/code. The warnings should be emitted if the old name is used with instructions for using the new name. Don't crash with an error. #3803 shows how this can be done.

**Comment by yaahc** [comment]
> Do we want to do this? Recently all internal uses of black/white-list were renamed in the compiler. No user facing things were renamed though. This would definitely be user facing, both the lint name and the configuration name.

I for one am in favor

**Comment by matthiaskrgr** [comment]
FTR this is the similar pr that was merged into rustc recently: https://github.com/rust-lang/rust/commit/1e567c11686f20b4e7422ba9cc0f1eac45e9c9fb

I also have not objections to merging this :) , but we should definitely have some kind of deprecation message! (As far as I can see there is none as of this version of this PR..?)

File: clippy_lints/src/blacklisted_name.rs [hunk 1]
```diff
@@ -17,33 +17,33 @@ declare_clippy_lint! {
...
+    pub DISALLOWED_NAME,
     style,
-    "usage of a blacklisted/placeholder name"
+    "usage of a disallowed/placeholder name"
 }
 
 #[derive(Clone, Debug)]
-pub struct BlacklistedName {
-    blacklist: FxHashSet<String>,
+pub struct DisAllowedName {
```
Comment (phansch) [inline clippy_lints/src/blacklisted_name.rs:26]: I think this should be 
```suggestion
pub struct DisallowedName {
```

**Comment by phansch** [comment]
I'm also supportive of merging this (once the lint renaming handling has been added)

**Comment by Ryan1729** [comment]
> It's got to be done without breaking peoples config/code. The warnings should be emitted if the old name is used with instructions for using the new name. Don't crash with an error. #3803 shows how this can be done.

I was curious about what changes in #3803 were relevant to ensuring these qualities. As far as I can tell, only the following files are relevant:

* [clippy_lints/src/utils/attrs.rs](https://github.com/rust-lang/rust-clippy/pull/3803/files#diff-41f8cc6a91e76e544ec320b689b84b92)
* [clippy_lints/src/utils/conf.rs](https://github.com/rust-lang/rust-clippy/pull/3803/files#diff-383721c39e0861b94506148273a3f66d)
* [tests/ui-toml/conf_deprecated_key/clippy.toml](https://github.com/rust-lang/rust-clippy/pull/3803/files#diff-b33ea52b1f170cce8868de833e5f33a2)
* [tests/ui-toml/conf_deprecated_key/conf_deprecated_key.rs](https://github.com/rust-lang/rust-clippy/pull/3803/files#diff-a409e445d7dbff37f881c2b5fe6b3650)
* [tests/ui-toml/conf_deprecated_key/conf_deprecated_key.stderr](https://github.com/rust-lang/rust-clippy/pull/3803/files#diff-7187a0a8e5db0c2f5e0da58945ce62fc)
* [tests/ui-toml/good_toml_no_false_negatives/clippy.toml](https://github.com/rust-lang/rust-clippy/pull/3803/files#diff-530857cc6085ea63ab60c1b7f6863009)
* [tests/ui-toml/good_toml_no_false_negatives/conf_no_false_negatives.rs](https://github.com/rust-lang/rust-clippy/pull/3803/files#diff-42d047ce073098b354aa7b864a6813a4)
* [tests/ui/renamed_builtin_attr.rs](https://github.com/rust-lang/rust-clippy/pull/3803/files#diff-5735d94e14ee1fee1a500e40554f0314)
* [tests/ui/renamed_builtin_attr.stderr](https://github.com/rust-lang/rust-clippy/pull/3803/files#diff-6bc9caf5694d7d0046381909f92f9ac9)


Hopefully that helps anyone else who would otherwise find themselves reading the entire diff.

**Comment by flip1995** [comment]
Alright, let's move forward with the renaming!

@mocsy 2 things have to be done:

1. Register the lint as renamed (use `register_renamed` for this)
2. Register the old configuration option as renamed. (See in `utils/conf.rs` how this was done with `cyclomatic_complexity` and do the same for this lint)

Each of the two steps require tests to be added. For 1. we already have a test file IIRC. For 2. add tests similar to the tests for `cognitive_complexity` linked above by @Ryan1729.

**Comment by mocsy** [comment]
> Alright, let's move forward with the renaming!
> 
> @mocsy 2 things have to be done:
> 
> 1. Register the lint as renamed (use `register_renamed` for this)
> 2. Register the old configuration option as renamed. (See in `utils/conf.rs` how this was done with `cyclomatic_complexity` and do the same for this lint)
> 
> Each of the two steps require tests to be added. For 1. we already have a test file IIRC. For 2. add tests similar to the tests for `cognitive_complexity` linked above by @Ryan1729.

All right, I'll do these steppes.

File: clippy_lints/src/disallowed_name.rs [hunk 2]
```diff
@@ -17,33 +17,33 @@ declare_clippy_lint! {
     /// ```rust
     /// let foo = 3.14;
     /// ```
-    pub BLACKLISTED_NAME,
+    pub DISALLOWED_NAME,
```
Comment (flip1995) [inline clippy_lints/src/disallowed_name.rs:20]: Let's rename it properly:

```suggestion
    pub DISALLOWED_NAMES,
```

Comment (mocsy) [reply clippy_lints/src/disallowed_name.rs:20]: I believe it's the lint definition which is in singular:
```rust
#![warn(clippy::blacklisted_name)]
```
not the toml entry:
```toml
blacklisted-names = ["toto", "tata", "titi"]
```
which is in plural.

**Comment by mocsy** [comment]
> There are tests missing for the old configuration name and the renaming of the lint.
> 
> Also some test files are still named `*blacklist*`, it would be great if you could rename them also.
These are done, please confirm this approach is good.

File: clippy_lints/src/disallowed_name.rs (hunk 2 above)
Comment (flip1995) [reply clippy_lints/src/disallowed_name.rs:20]: Yeah, but it should be in plural, as per our [lint naming conventions](https://rust-lang.github.io/rfcs/0344-conventions-galore.html#lints) (point 3)

File: tests/ui/blacklisted_name.stderr [hunk 3]
```diff
@@ -1,88 +1,16 @@
-error: use of a blacklisted/placeholder name `foo`
-  --> $DIR/blacklisted_name.rs:11:9
+error: lint `clippy::blacklisted_name` has been renamed to `clippy::disallowed_name`
+  --> $DIR/blacklisted_name.rs:9:9
    |
-LL | fn test(foo: ()) {}
-   |         ^^^
+LL | #![warn(clippy::blacklisted_name)]
+   |         ^^^^^^^^^^^^^^^^^^^^^^^^ help: use the new name: `clippy::disallowed_name`
```
Comment (flip1995) [inline tests/ui/blacklisted_name.stderr:5]: Please include the rename tests in `tests/ui/rename.rs`

**Review by flip1995 — COMMENTED** [review]
There is still a test missing for the renamed configuration option (or did I miss this test?)

File: tests/ui/blacklisted_name.stderr (hunk 3 above)
Comment (mocsy) [reply tests/ui/blacklisted_name.stderr:5]: Okay, shall I remove `tests/ui/blacklisted_name ` then?

Comment (flip1995) [reply tests/ui/blacklisted_name.stderr:5]: Yes, after you have one `#[warn(clippy::blacklisted_name)]` in the rename test file you can remove this.

**Comment by flip1995** [comment]
ping from triage @mocsy. Anything I can help you with to get this over the finish line?

**Comment by giraffate** [comment]
@flip1995 2 weeks have passed with no activity from previous ping, so this PR should be closed according to Rust triage procedure.

**Pull request closed without merging**